pkg/signature/kms/hashivault/client.go

```diff
@@ -313,7 +313,27 @@ func (h hashivaultClient) verify(sig, digest []byte, alg crypto.Hash, opts ...si
 			if h.keyVersion > 0 {
 				vaultDataPrefix = fmt.Sprintf("vault:v%d:", h.keyVersion)
 			} else {
-				vaultDataPrefix = vaultV1DataPrefix
+				// use hashivault client to grab the latest version of the key and use that as the prefix
```

Comment:
I have one concern with this implementation - it ignores the `keyCache` completely; I'd rather have it invalidate the cache, call `public()` (which would hydrate the cache with the `latest_version`. 

It still results in another round trip to Vault to do the verification, but we at least have the key material for re-use in memory.